### PR TITLE
[#163] Implement Post input component

### LINKED ISSUE #163: Implement Post input component
**Is your feature request related to a problem? Please describe.**

Post input does not check for character count, if the character count exceeds the backend supported count, the backend would error and the post input does not give the user any feedback.

Post input does not support hashtags or tagging people.

image

**Describe the solution you'd like**

Post input should have a character count and limit the post length.

When tagging a user using `@handle` or a topic `#topic` the frontend should check if the handle or topic is already existing and if so change it to a hyperlink. 

**Describe alternatives you've considered**
Use backend to change to hyperlink however it is easier to do in the frontend

**Additional context**

### PR DESCRIPTION
**Pull request by scorpionknifes**
# Description:
- Changed font to Poppins 
- Created Post Input Component
	- the component uses a div instead of an input (required to render unsafe HTML in content)
	- custom cursor location (since not an input need to use custom code to save cursor position during state change)
- Added Post Input to the Page
- Added Global Context for Tags and Handles (not connected to backend yet)

| Figma Design | Implemented |
| - | - |
| image | image |

Closes #163

No testing is done yet.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: frontend/src/index.jsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { BrowserRouter } from 'react-router-dom'
 import TimeAgo from 'javascript-time-ago'
 import en from 'javascript-time-ago/locale/en.json'
 import App from './App'
-import '@fontsource/roboto'
+import '@fontsource/poppins'
```
Comment (scorpionknifes) [inline frontend/src/index.jsx:7]: eslint failing for no apparent reason? ignore?

Comment (uoa-lzha989) [reply frontend/src/index.jsx:7]: I think if there's no clear reason for eslint to be failing, then happy to put an ignore and/or flag for investigation down the line if it happens again.

**Review by uoa-lzha989 — APPROVED** [review]
Aside from that eslint issue, everything is looking pretty good.

**Review by colaMeowSY — APPROVED** [review]
LGTM

**Pull request merged**